packages/core/src/browser/tree/tree-model.ts

```diff
@@ -302,21 +302,22 @@ export class TreeModelImpl implements TreeModel, SelectionProvider<ReadonlyArray
         return iterator && this.doGetNextNode(iterator);
     }
 
-    protected doGetNextNode(iterator: TreeIterator): SelectableTreeNode | undefined {
+    protected doGetNextNode(iterator: IterableIterator<TreeNode>): SelectableTreeNode | undefined {
```

Comment:
I'd rather just change this to be `extends IterableIterator<TreeNode>`, but that might be slightly more breaking. Anyone who's implemented an iterator that isn't iterable would probably be grateful in the long run, though. Who wants to type `.next()` all the time?

https://github.com/eclipse-theia/theia/blob/210fa4ac6875716f23ac70f94fe6f0ffc693bfe5/packages/core/src/browser/tree/tree-iterator.ts#L20